Vector/ViewController/HomeViewController.m

```diff
@@ -245,12 +257,36 @@ - (void)viewDidLayoutSubviews
 - (void)showAuthenticationScreen
 {
     [[AppDelegate theDelegate] restoreInitialDisplay:^{
-        
+
         [self performSegueWithIdentifier:@"showAuth" sender:self];
-        
+
     }];
 }
 
+- (void)showAuthenticationScreenWithNextLinkParameters:(NSDictionary *)nextLinkParameters
+{
+    if (self.authViewController)
+    {
+        NSLog(@"[HomeViewController] Universal link: Forward next_link parameter to the existing AuthViewController");
+        [self.authViewController registerWithNextLinkParameters:nextLinkParameters];
+    }
+    else
+    {
+        NSLog(@"[HomeViewController] Universal link: Logout current sessions and open AuthViewController to complete the registration in next_link");
+
+        // Logout out before opening the Authentication screen
+        [[AppDelegate theDelegate] logout];
+
+        authViewControllerNextLinkParameters = nextLinkParameters;
+
+        [[AppDelegate theDelegate] restoreInitialDisplay:^{
+
+            [self performSegueWithIdentifier:@"showAuth" sender:self];
+            
+        }];
```

Comment:
[_homeViewController showAuthenticationScreen] is already called during logout

restoreInitialDisplay and performSegueWithIdentifier are useless.